app/redux/createIpcAction.ts

```diff
@@ -0,0 +1,23 @@
+import { ActionReducerMapBuilder, AsyncThunk, AsyncThunkPayloadCreator, createAsyncThunk } from '@reduxjs/toolkit';
+
+type EmptyObject = Record<string, never>;
+type IdentityFn<T> = (arg0: T) => T;
+
+const DEFAULT_IPC_REDUCERS = {
+  onFulfilled: (state, action) => ({ ...state, ...action.payload }),
+  onPending: (state) => state,
+  onRejected: (state, action) => ({ ...state, error: action.payload }),
+};
+
+const createIpcAction = <T>(
+  typePrefix: string,
+  payloadCreator: AsyncThunkPayloadCreator<T>,
+  { onFulfilled, onRejected, onPending } = DEFAULT_IPC_REDUCERS
+): [AsyncThunk<T, void, EmptyObject>, (builder: ActionReducerMapBuilder<T>) => ActionReducerMapBuilder<T>] => {
```

Comment:
the return type is possible to separate. Not unique to the specific createIpcAction function types, they are universal.